tests/validation/tests/v3_api/common.py

```diff
@@ -540,22 +540,31 @@ def wait_until_active(url, timeout=120):
     return
 
 
-def validate_http_response(cmd, target_name_list, client_pod=None):
+def validate_http_response(cmd, target_name_list, client_pod=None, max_retries=60):
     if client_pod is None and cmd.startswith("http://"):
         wait_until_active(cmd, 60)
     target_hit_list = target_name_list[:]
     count = 5 * len(target_name_list)
+    result = "503"
+    retry = 0
     for i in range(1, count):
+        if retry > max_retries:
+            break
+        retry = retry + 1
+
         if len(target_hit_list) == 0:
             break
-        if client_pod is None:
-            curl_cmd = "curl " + cmd
-            result = run_command(curl_cmd)
-        else:
-            wget_cmd = "wget -qO- " + cmd
-            result = kubectl_pod_exec(client_pod, wget_cmd)
-            result = result.decode()
-        result = result.rstrip()
+        while result.find("503") != -1:
```

Comment:
Instead of writing this here, let's add a helper that can be called by `validate_ingress` to do the initial check of the endpoint.